engines/buried/graphics.cpp

```diff
@@ -0,0 +1,769 @@
+	    REQUIRED(0x00FC),     REQUIRED(0x00FD),     REQUIRED(0x00FE),     REQUIRED(0x00FF)
+};
+
+#undef REQUIRED
+#undef NOT_REQUIRED
+
+#endif // USE_ICONV
+
+Graphics::Font *GraphicsManager::createFont(int size, bool bold) const {
+	// MS Gothic for the Japanese version
+	// Arial or Arial Bold for everything else
+	if (_vm->getLanguage() == Common::JA_JPN)
+		return createMSGothicFont(size, bold);
+
+	return createArialFont(size, bold);
+}
+
+Graphics::Font *GraphicsManager::createArialFont(int size, bool bold) const {
+	Common::String defaultBaseName = bold ? "arialbd.ttf" : "arial.ttf";
+
+	Common::SeekableReadStream *stream = SearchMan.createReadStreamForMember(defaultBaseName);
+
+	// Map the heights needed to point sizes
+	if (bold) {
+		if (size != 20)
+			error("Unhandled Arial Bold height %d", size);
+
+		size = 12;
+	} else {
+		switch (size) {
+		case 12:
+		case 13:
+			size = 7;
+			break;
+		case 14:
+			size = 8;
+			break;
+		default:
+			error("Unhandled Arial height %d", size);
+		}
+	}
+
+	// TODO: Make the monochrome mode optional
+	// Win3.1 obviously only had raster fonts, but BIT Win3.1 will render
+	// with the TrueType font on Win7/Win8 (at least)
+	// FIXME: The font is slightly off from the original... need to check. Sizes are right though!
+
+	// Enable code page mapping only if we don't have iconv. Otherwise, we'll
+	// let that handle mapping for us.
+#ifdef USE_ICONV
+	static const uint32 *codePageMapping = nullptr;
+#else
+	static const uint32 *codePageMapping = s_codePage1252;
+#endif
+
+	Graphics::Font *font;
+
+	if (stream) {
+		font = Graphics::loadTTFFont(*stream, size, Graphics::kTTFSizeModeCharacter, 96, _vm->isTrueColor() ? Graphics::kTTFRenderModeLight : Graphics::kTTFRenderModeMonochrome, codePageMapping);
+
+		delete stream;
+	} else {
+		const char *fname;
+		if (bold)
+			fname = "LiberationSans-Bold.ttf";
+		else
+			fname = "LiberationSans-Regular.ttf";
+
+		font = Graphics::loadTTFFontFromArchive(fname, size, Graphics::kTTFSizeModeCharacter, 96, _vm->isTrueColor() ? Graphics::kTTFRenderModeLight : Graphics::kTTFRenderModeMonochrome, codePageMapping);
+	}
+
+	if (!font)
+		error("Failed to load Arial%s font", bold ? " Bold" : "");
+
+	return font;
+}
+
+Cursor GraphicsManager::setCursor(Cursor newCursor) {
+	// Don't set the cursor again
+	if (newCursor == _curCursor)
+		return _curCursor;
+
+	Cursor oldCursor = _curCursor;
+	Graphics::Cursor *cursor = nullptr;
+	Graphics::WinCursorGroup *cursorGroup = nullptr;
+
+	if (newCursor == kCursorArrow) {
+		cursor = Graphics::makeDefaultWinCursor();
+	} else if (newCursor == kCursorWait) {
+		cursor = Graphics::makeBusyWinCursor();
+	} else {
+		cursorGroup = _vm->_mainEXE->getCursorGroup(newCursor);
+
+		if (!cursorGroup)
+			return kCursorNone;
+
+		cursor = cursorGroup->cursors[0].cursor;
+	}
+
+	if (!cursor)
+		error("Failed to find cursor %d", newCursor);
+
+	// TODO: Fallback mode for platforms without cursor palettes in 8bpp mode?
+
```

Comment:
As far as I can tell, there aren't any official platforms remaining that don't support cursor palettes.

```suggestion
```